src/fileio.cpp

```diff
@@ -135,7 +134,7 @@ bool FioCheckFileExists(const std::string &filename, Subdirectory subdir)
  */
 bool FileExists(const std::string &filename)
 {
-	return access(OTTD2FS(filename).c_str(), 0) == 0;
+	return std::filesystem::exists(OTTD2FS(filename));
```

Comment:
Do we want an exception being thrown "on underlying OS API errors"? So, essentially crash the game?

Or do we want to get the non-throwing overload and return `false` when things go wrong?